src/views/forms/AccountManagementFormsNew/WithdrawForm/WithdrawButtonAndReceipt.tsx

```diff
@@ -0,0 +1,192 @@
+import { useState } from 'react';
+
+import { Asset, RouteResponse } from '@skip-go/client';
+import { shallowEqual } from 'react-redux';
+import styled from 'styled-components';
+import tw from 'twin.macro';
+import { formatUnits } from 'viem';
+
+import { ButtonAction, ButtonSize, ButtonType } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+import { NumberSign, TOKEN_DECIMALS } from '@/constants/numbers';
+
+import { ConnectionErrorType, useApiState } from '@/hooks/useApiState';
+import { useStringGetter } from '@/hooks/useStringGetter';
+import { useTokenConfigs } from '@/hooks/useTokenConfigs';
+
+import { Button } from '@/components/Button';
+import { Details } from '@/components/Details';
+import { DiffOutput } from '@/components/DiffOutput';
+import { Output, OutputType } from '@/components/Output';
+import { Tag } from '@/components/Tag';
+import { WithReceipt } from '@/components/WithReceipt';
+import { WithTooltip } from '@/components/WithTooltip';
+import { OnboardingTriggerButton } from '@/views/dialogs/OnboardingTriggerButton';
+
+import { calculateCanAccountTrade } from '@/state/accountCalculators';
+import { getSubaccount } from '@/state/accountSelectors';
+import { useAppSelector } from '@/state/appTypes';
+
+import { isTruthy } from '@/lib/isTruthy';
+
+import { RouteWarningMessage } from '../RouteWarningMessage';
+
+type ElementProps = {
+  withdrawToken?: Asset;
+  route?: RouteResponse;
+
+  isDisabled?: boolean;
+  isLoading?: boolean;
+};
+
+export const WithdrawButtonAndReceipt = ({
+  withdrawToken,
+  route,
+
+  isDisabled,
+  isLoading,
+}: ElementProps) => {
+  const stringGetter = useStringGetter();
+
+  const { leverage } = useAppSelector(getSubaccount, shallowEqual) ?? {};
+
+  const canAccountTrade = useAppSelector(calculateCanAccountTrade, shallowEqual);
+  const { usdcDecimals } = useTokenConfigs();
+  const { connectionError } = useApiState();
+
+  const fees = Number(route?.usdAmountIn) - Number(route?.usdAmountOut);
+  const submitButtonReceipt = [
+    {
+      key: 'amount-inputted',
+      label: <span>Amount</span>,
+      value: (
+        <Output
+          type={OutputType.Asset}
+          value={
+            route?.amountIn
+              ? formatUnits(
+                  BigInt(route.amountIn ?? '0'),
+                  withdrawToken?.decimals ?? usdcDecimals
```

Comment:
i would again not default to 6 decimals here, that feels wrong in almost every case where withdrawToken.decimals isnt defined 😬 throw an error instead if .decimals isnt defined?

Reply:
hmm the amount in will always be USDC but yeah i'm down with this

Reply:
while we're at it we should probably also throw an error if withdrawToken.decimals !== usdcDecimals.

Reply:
oh are we only doing usdc withdrawals? 

also usdc is 18 decimals on some chains LOL